compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -1077,6 +1077,9 @@ impl<'tcx> MutVisitor<'tcx> for Integrator<'_, 'tcx> {
                 *target = self.map_block(*target);
                 *unwind = self.map_unwind(*unwind);
             }
+            TerminatorKind::TailCall { .. } => {
+                // FIXME(explicit_tail_calls): figure out how exactly tail calls are inlined
```

Comment:
yea this is gonna be an issue... maybe we should just make the cost of tail calls infinite for now, so that functions with tail calls inside don't get inlined?

Reply:
Made it so cost inliner errors if callee has tail calls and changes fixmes a little: 7727481